src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerDelegate.scala

```diff
@@ -176,6 +176,12 @@ private[tracker] class InstanceTrackerDelegate(
           }(ExecutionContexts.callerThread)
           instanceTrackerRef.tell(InstanceTrackerActor.Subscribe, ref)
           NotUsed
+      }.prefixAndTail(1).map {
+        case (Seq(el: InstancesSnapshot), rest) =>
+          // The contract is that all messages delivered after the first snapshot message will be instance change
+          (el, rest.asInstanceOf[Source[InstanceChange, NotUsed]]): @silent
+        case _ =>
+          throw new IllegalStateException("First message expected to be an InstancesSnapshot; this is a bug")
```

Comment:
do you mind printing what we actually received? If it was no element of element of a different type... ?